core/pkg/utils/files.go

```diff
@@ -18,6 +20,31 @@ func FileExists(path string) (bool, error) {
 	return true, nil
 }
 
+func CopyFile(src, dst string) error {
```

Comment:
would be nice to have a doc comment describing prereqs and any nuances

```
// CopyFile copies the contents of `src` into `dst`.
//
// The source file must exist, and the destination file's parent folder must exist,
// or else this is a no-op and an error is returned.
//
// This operation is not atomic: if the source file is modified during the copy,
// the destination file's contents may be corrupted.
```

not sure about `the destination file's parent folder must exist` (all the more reason to document it!)

Reply:
Actually, I would just as soon create all parent directories, I basically never want to fail a copy because a parent doesn't exist. Assuming you agree I'll update the function to also create parent directories.